### PR TITLE
MBL-1211: Pass down pledge total to thank you page for new page copy

### PR DESCRIPTION
**Pull request by amy-at-kickstarter**
# 📲 What

Pass the pledge total into the thank you page.

# 🤔 Why

The new string for post-campaign backings on the thank-you page will reference the pledge amount, so we need it in the view model.

### CODE REVIEW

File: Library/ViewModels/ThanksViewModel.swift [hunk 1]
```diff
@@ -77,21 +78,25 @@ public final class ThanksViewModel: ThanksViewModelType, ThanksViewModelInputs,
...
       .skipNil()
       .map { ($0.reward, $0.checkoutData) }
 
-    self.backedProjectText = project.map {
-      let string = Strings.You_have_successfully_backed_project_html(
-        project_name: $0.name
-      )
+    self.backedProjectText = self.configureWithDataProperty.signal
+      .skipNil()
+      .map { project, _, _, _ in
```
Comment (amy-at-kickstarter) [inline Library/ViewModels/ThanksViewModel.swift:89]: Fourth arg here will be the pledge total.

**Comment by amy-at-kickstarter** [comment]
Whoops, converting to draft while I fix tests.

File: Library/ViewModels/PledgeViewModel.swift [hunk 2]
```diff
@@ -905,7 +905,7 @@ public class PledgeViewModel: PledgeViewModelType, PledgeViewModelInputs, Pledge
         isApplePay: isApplePay
       )
 
-      return (data.project, baseReward, checkoutPropsData)
+      return (data.project, baseReward, checkoutPropsData, data.pledgeTotal)
```
Comment (amy-at-kickstarter) [inline Library/ViewModels/PledgeViewModel.swift:908]: One line to change it, another 40 lines to fix the tests about it 😆

**Review by scottkicks — CHANGES_REQUESTED** [review]
oops disregard this

File: Library/ViewModels/ThanksViewModel.swift (hunk 1 above)
Comment (scottkicks) [reply Library/ViewModels/ThanksViewModel.swift:89]: Is the plan to then gate the string from within this output?

Comment (amy-at-kickstarter) [reply Library/ViewModels/ThanksViewModel.swift:89]: @scottkicks Correct; this is just plumbing to get all the data where it needs to go.

**Pull request merged**